AWSPinpoint/AWSPinpointEventRecorder.m

```diff
@@ -198,10 +207,17 @@ - (NSString *)validateOrRetrieveSessionId:(NSString *) sessionId {
     }
     NSData *sessionData = [self.context.configuration.userDefaults dataForKey:AWSPinpointSessionKey];
     if (sessionData) {
-        AWSPinpointSession *previousSession = [NSKeyedUnarchiver unarchiveObjectWithData:sessionData];
+        NSError *error;
+        AWSPinpointSession *previousSession = [AWSNSCodingUtilities versionSafeUnarchivedObjectOfClass:[AWSPinpointSession class]
+                                                                                              fromData:sessionData
+                                                                                                 error:&error];
+        if (error || !previousSession) {
```

Comment:
`!previousSession` is not an error in restoring the archive right? It can mean that we do not have an archived object. So, should we just have ` if (error) { ... }`

Reply:
If we didn't have an archived object, we shouldn't have been able to hit this code in the first place, since we check for the existence of `sessionData` on L209 above. Thus, if we have retrieved session data, but are unable to unarchive it, it implies the archived data was corrupted and we should follow the error path.

Practically speaking, it's no change in behavior, since you can see below that the very next thing we try is to return `previousSession.sessionId`, returning the default if it's not present.